tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,168 @@ func runRacer(eps []string, round int) {
+	noOfPrefixes := 20              // total number of prefixes which will be watched upon
+	watchPerPrefix := 10            // number of watchers per prefix
+	reqRate := 20                   // put request per second
+	keyPrePrefix := 1000            // max number of keyPrePrefixs for put operation
+
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	prePrefixKeys := generateRandomKeys(10, keyPrePrefix)
+	// map for holding prefixes and values
+	prefixKeysMap := make(map[string][]string)
+	// actual keys for performing put operation
+	keys := make([]string, 0)
+
+	// initialize the map with all prefixes
+	for _, prefix := range prefixes {
+		prefixKeysMap[prefix] = make([]string, 0)
+	}
+
+	// preapre keys and maintain the list of keys for each prefix
+	prefixIndex := 0
+	for _, preKey := range prePrefixKeys {
+		prefix := prefixes[prefixIndex]
+		key := prefix + preKey
+		keys = append(keys, key)
+
+		// add all possible updates on prefixes for the generated key
+		for prefixkey, vals := range prefixKeysMap {
+			if strings.HasPrefix(key, prefixkey) {
+				prefixKeysMap[prefixkey] = append(vals, key)
+			}
+		}
+		prefixIndex = (prefixIndex + 1) % len(prefixes)
+	}
+
+	var (
+		revision int64
+		wg       sync.WaitGroup
+		pr       *clientv3.PutResponse
+		gr       *clientv3.GetResponse
+		err      error
+	)
+
+	// create client for performing get and put operations
+	ctx := context.Background()
+	client := randClient(eps)
+	defer client.Close()
+
+	// get revision using get request
+	gr, err = client.Get(ctx, "non-existant")
+	if err != nil {
+		log.Fatal("Error occured while trying to get the revision.")
+	}
+	revision = gr.Header.Revision
+
+	ctxt, cancel := context.WithDeadline(ctx, time.Now().Add(runningTime))
+	defer cancel()
+
+	// generate and put keys in cluster
+	go func() {
+		limiter := rate.NewLimiter(rate.Limit(reqRate), reqRate)
+		for _, key := range keys {
+			// limit key put as per reqRate
+			err = limiter.Wait(ctxt)
+			if err != nil {
+				break
+			}
+
+			// perform the put operation
+			pr, err = client.Put(ctxt, key, key)
+			if err == context.DeadlineExceeded {
+				break
+			}
+
+			if err != nil {
+				log.Printf("Error: %v occured while trying to key: %v, value : %v to kv store.\n", err, key, key)
+				continue
+			}
+		}
+	}()
+
+	// create watches on prefixes and compare their values
+	wg.Add(noOfPrefixes * watchPerPrefix)
+	for i := 0; i < noOfPrefixes; i++ {
+		prefix := prefixes[i]
+		for j := 0; j < watchPerPrefix; j++ {
+			// for all watcher verify the watch out put with actual key updates on prefixes
+			go func(prefix string) {
+				defer wg.Done()
+				rc := randClient(eps)
+				defer rc.Close()
+				wc := rc.Watch(ctx, prefix, clientv3.WithPrefix(), clientv3.WithRev(revision))
+				// get all keys expected for the prefix
+				keys := prefixKeysMap[prefix]
+
+				index := 0
```

Comment:
remove this line. also we need to update i at line 332. i suggest to name the looping variable to n since we already use i at line 307.